src/result.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+export type Result<V, E> = Success<V, E>| Failure<V, E>;
+
+export namespace Result {
```

Comment:
Yeah I think so. Some of this API isn't super useful or idiomatic though, will need a bit of iteration I expect.